62_CAD/shaders/main_pipeline/fragment_shader.hlsl

```diff
@@ -408,11 +410,76 @@ float4 fragMain(PSInput input) : SV_TARGET
     float localAlpha = 0.0f;
     float3 textureColor = float3(0, 0, 0); // color sampled from a texture
 
-    // TODO[Przemek]: Disable All the object rendering paths if you want.
     ObjectType objType = input.getObjType();
     const uint32_t currentMainObjectIdx = input.getMainObjectIdx();
     const MainObject mainObj = mainObjects[currentMainObjectIdx];
-    
+
+    // TRIANGLE RENDERING
+    {
+        float3 v0 = input.getScreenSpaceVertexPos(0);
+        float3 v1 = input.getScreenSpaceVertexPos(1);
+        float3 v2 = input.getScreenSpaceVertexPos(2);
+
+        float2 start;
+        float2 end;
+        const float3 baryCoord = nbl::hlsl::spirv::BaryCoordKHR;
+
+        // TODO: figure out if branching can be reduced
+        // finding line start and end points by excluding vertex with the lowest barycentric coordinate value
```

Comment:
you'll need to do SDF with potentially all 3 edges for the outline:
I've already covered this in our chat [here](https://discord.com/channels/593902898015109131/1338454683869515847/1351406137827921961)

```cpp
float triangleSDF = MAX_DISTANCE;
for (int n = 0; n < 3; ++n)
{
  float distanceToEdge = screenSpaceHeight[n]*baryCoord[n];
  float edgeSDF = abs(distanceToEdge) - thickness; 
  const bool edgeCanAffectCurrentPixel = edgeSDF <= aaFactor; 

  if (complexStyle && edgeCanAffectCurrentPixel)
  {
    edgeSDF = ...; // do complex line style with the custom lineStyleClipper and shit, you know the else case thing
  }

  triangleSDF = min(triangleSDF, edgeSDF);
}
```

Reply:
outdated, resolving